bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/handlers/HeapDumpAnalysisHandler.java

```diff
@@ -0,0 +1,123 @@
+/*******************************************************************************
+ * Copyright 2021 Espressif Systems (Shanghai) PTE LTD. All rights reserved.
+ * Use is subject to license terms.
+ *******************************************************************************/
+package com.espressif.idf.ui.handlers;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.core.commands.ExecutionEvent;
+import org.eclipse.core.commands.ExecutionException;
+import org.eclipse.core.expressions.IEvaluationContext;
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.ui.IWorkbenchWindow;
+import org.eclipse.ui.PartInitException;
+import org.eclipse.ui.ide.IDE;
+import org.eclipse.ui.part.FileEditorInput;
+
+import com.espressif.idf.core.IDFCorePlugin;
+import com.espressif.idf.core.IDFEnvironmentVariables;
+import com.espressif.idf.core.ProcessBuilderFactory;
+import com.espressif.idf.core.logging.Logger;
+import com.espressif.idf.core.util.FileUtil;
+import com.espressif.idf.core.util.IDFUtil;
+import com.espressif.idf.ui.tracing.HeapTracingAnalysisEditor;
+import com.espressif.idf.ui.update.AbstractToolsHandler;
+
+/**
+ * Handler class for handling the context menu action
+ * 
+ * @author Ali Azam Rana
+ *
+ */
+public class HeapDumpAnalysisHandler extends AbstractToolsHandler
```

Comment:
Hmm..do you see the need to extend AbstractToolsHandler instead AbstractHandler? If it's only for console purposes I would suggest using the new IDFConsole().getConsoleStream()

Reply:
Done